### PR TITLE
Sign In - Sign Up - Auth

### PR DESCRIPTION
**Pull request by horsemaker**
- Added Auth Context
- Wrapped Auth Provider
- Added Auth Services
- Added Sign In Page
- Added Sign Up Page

**Sign In**

recording

**Sign Up**

recording

### CODE REVIEW

**Review by narharikale — APPROVED** [review]
clean code 
good folder structure
and great error handling

File: src/frontend/pages/SignInPage/SignInPage.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
+.auth-screen {
+  margin-left: 1rem;
+  margin-right: 1rem;
```
Comment (MiheerTamkhane) [inline src/frontend/pages/SignInPage/SignInPage.css]: You can use margin : 0 1rem; here

**Review by MiheerTamkhane — APPROVED** [review]
Code is written well, Structured well and also loved the naming convention.
Nice work.

**Review by t007rushi — APPROVED** [review]
Liked long but clear naming 
Nicely modules the code
Go ahead

File: src/frontend/pages/SignInPage/SignInPage.css (hunk 1 above)
Comment (horsemaker) [reply src/frontend/pages/SignInPage/SignInPage.css]: I have also used `margin-top`. I will use the short-hand property here. Thank you!

**Pull request merged**